### PR TITLE
Token vefiry 86712102

### PR DESCRIPTION
**Pull request by mikrowelt**
added token verification
added revoke token
improved docs
added a new example
added events support
added token auto signin
fixed bugs

works only with sandbox1 api until merged to staging

https://www.pivotaltracker.com/story/show/86712102

### CODE REVIEW

**Comment by mikrowelt** [comment]
this is a requireemnt for merge https://github.com/openedinc/openedengine/pull/1392

File: index.html [hunk 1]
```diff
@@ -68,24 +68,30 @@
...
     </div>
 	<script>
-	  $('#login').click(function(){
-		window.OpenEd.api.init({
-		  client_id: '66d0354299f860b4e39c56979f5cfd7d5216f67ce958c19ad5a3a68e6b0a2399',
-		  redirect_uri: 'http://oauth2.openedtest.com:4200/callback.html'
-	    });
-	    window.OpenEd.api.login(function () {
-		  window.OpenEd.api.request('/users/me.json', null, function (data) {
+      //Initing the API
```
Comment (artursmirnov) [inline index.html]: I would say "initializing"

File: oauth.js [hunk 2]
```diff
@@ -8,26 +8,59 @@
 
     tokenPrefix: '_openEd',
 
-    isInited: false,
+    apiHost: 'https://api-staging.opened.io',
```
Comment (artursmirnov) [inline oauth.js]: I thought we desided to use production urls, not staging ones.

File: oauth.js [hunk 3]
```diff
@@ -8,26 +8,59 @@
...
 
-    apiHost: 'https://api.opened.io',
+    openedHost: 'https://staging.opened.io',
```
Comment (artursmirnov) [inline oauth.js]: The same about staging. I would say, we shouldn't promote staging to other devs.

File: oauth.js [hunk 4]
```diff
@@ -96,53 +169,91 @@
...
+    },
+
+    checkTokenDate: function () {
+      var tokenDate = new Date(parseInt(localStorage.getItem(this.tokenPrefix + '.expires_in')));
+      var now = this.now();
+      return now.getTime() < tokenDate.getTime();
     },
-    /*
+
+    expairDate: function (expairsIn) {
```
Comment (artursmirnov) [inline oauth.js]: Here is a typo: Expire

File: oauth.js [hunk 5]
```diff
@@ -96,53 +169,91 @@
...
+      var date = this.now();
+      date.setTime(date.getTime() + (parseInt(expairsIn) * 1000) );
```
Comment (artursmirnov) [inline oauth.js]: Same typo. Please check it over all the code

**Comment by artursmirnov** [comment]
@mikrowelt I'm done with review. It's fine in general, just a few small issues. Please merge once it's fixed.

**Pull request merged**